### PR TITLE
#1640 grade document

### PR DESCRIPTION
**Pull request by fidoriel**
fix for #1640 grade document

using the same 'final' flag as the upload_grades func

### CODE REVIEW

File: evap/grades/templates/grades_course_view.html
Comment (He3lixxx) [inline evap/grades/templates/grades_course_view.html:33]: The view that we are linking to here will get the `grade_document.id` passed in and load the `GradeDocument` from the database anyway. Thus, we could just do all the logic in the view, and would not require the additional parameter -- or am I missing something?

File: evap/grades/views.py
Comment (He3lixxx) [inline evap/grades/views.py:168]: This could then become
```
final_grades = grade_document.type == GradeDocument.Type.FINAL_GRADES
```
which would properly utilize the constant enum-like names we have for the two types without relying on whatever string is used in the database.

**Review by He3lixxx — CHANGES_REQUESTED** [review]
Thanks again!

We probably also want to have a small test for this.

**Comment by fidoriel** [comment]
I added the requested changes, now it works only with the bool flag final_grades.

File: evap/grades/templates/grades_course_view.html
Comment (He3lixxx) [reply evap/grades/templates/grades_course_view.html:33]: This change here is still part of the PR -- I guess you didn't add the file before committing?

File: evap/grades/templates/grades_course_view.html
Comment (niklasmohrin) [inline evap/grades/templates/grades_course_view.html:35]: These two lines are almost the same, you could also inline the `if` into the url:
```django
href="{% url 'grades:edit_grades' semester.id course.id grade_document.id %}?final={% if final_grades %}true{% else %}false{% end %}"
```

Maybe you could even just render variable into a string using `{{ final_grades }}`. On that note: The variable should probably be called `is_final_grades` or similar to suggest it is a boolean and not a collection of the actual grades.

File: evap/grades/tests.py
Comment (niklasmohrin) [inline evap/grades/tests.py:140]: This can be shortened to
```python
baker.make(Evaluation, course__pk=1, course__semester__pk=1, state=...)
```
Additionally, the `grade_documents_are_deleted` is `False` by default

File: evap/grades/tests.py
Comment (niklasmohrin) [inline evap/grades/tests.py:149]: You use the id `1` again here, I suppose this is not what you intended?

**Review by niklasmohrin — CHANGES_REQUESTED** [review]

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

File: evap/grades/tests.py
Comment (niklasmohrin) [inline evap/grades/tests.py:254]: You could save this into a variable `url` once and make the other two uses shorter, your call :)

**Review by niklasmohrin — COMMENTED** [review]
Not much to criticize about the code :)

After rebasing on current `main`, CI should pass

File: evap/grades/tests.py
Comment (fidoriel) [reply evap/grades/tests.py:254]: I think this way ist shows what it does because it changes settings in between.

**Review by He3lixxx — APPROVED** [review]
Personally, I'd remove both comments from the test and store the url in a local variable as @niklasmohrin suggested:
```python3
url = f"/grades/semester/{grade_document.course.semester.pk}/course/{grade_document.course.pk}/edit/{grade_document.pk}"
# ...
response = self.app.get(url, user=grade_publisher)
```

But I think the current version does the job as well.

**Review by niklasmohrin — APPROVED** [review]
what richard said

**Comment by He3lixxx** [comment]
Alrighty so @fidoriel currently the formatting test fails, but I think that's due to a mistake we had in main. If you merge from or rebase onto upstream/main this should resolve.

If you want, you can apply what I suggested in https://github.com/e-valuation/EvaP/pull/1661#pullrequestreview-820163332.

Otherwise, we can merge this once all checks pass.

**Pull request merged**
